### PR TITLE
Dl/cypress@12 expect fix

### LINKED ISSUE #234: Interactors don't work after updating to Cypress 12
image

### PR DESCRIPTION
**Pull request by wKich**
## Motivation

Fix #234 

## Approach

It seems Cypress@12 doesn't expose `_commands` field https://github.com/cypress-io/cypress/blob/ae4cf2b553255f38ca83e14a09dd891b487e1a39/packages/driver/src/cypress/commands.ts#L45 which cause the error mentioned in #234

### CODE REVIEW

File: packages/with-cypress/src/cypress.ts [hunk 1]
```diff
@@ -39,37 +39,40 @@ function interact(interactions: Interaction<Element>[], command: CypressCommand)
...
     Cypress.Commands.add('expect', interactionExpect);
   }
   catch (e) {}
-  // @ts-expect-error Cypress stores a reference to commands object and use it to check overwiritability of commands
-  // https://github.com/cypress-io/cypress/blob/d378ec423a4a2799f90a6536f82e4504bc8b3c9e/packages/driver/src/cypress/commands.ts#L155
-  Cypress.Commands._commands['expect'] = interactionExpect;
-  // NOTE: Add interaction assertion function, Cypress also overrides `expect` method to a wrapper function
-  Cypress.Commands.overwrite('expect', interactionExpect);
+  if ('_commands' in Cypress.Commands) {
+    // @ts-expect-error Cypress stores a reference to commands object and use it to check overwiritability of commands
```
Comment (zburke) [inline packages/with-cypress/src/cypress.ts:64]: pedantic: `s/overwiritability/overwritability/`

Comment (wKich) [reply packages/with-cypress/src/cypress.ts:64]: Thanks :)

**Review by cowboyd — APPROVED** [review]
Good to go! Although it looks like it's not passing on CI.

**Review by ostapwd — APPROVED** [review]

**Comment by ostapwd** [comment]
Hi @wKich!
It looks like **cypres.json** should be replaced by ### cypress.config.ts in /packages/with-cypress dir

image

**Comment by cowboyd** [comment]
@wKich Odd that this is still failing Any idea what's up?

**Comment by wKich** [comment]
@cowboyd yep. It's time to fix https://github.com/cypress-io/cypress/issues/22733

**Comment by ostapwd** [comment]
Hi @cowboyd @wKich 
In my team, we are using Interactors for the FOLIO project, and we need the latest version of the Cypress. As a workaround is there a chance to add a new command instead of overriding existing, somethings like 'expectThat' (cy.expectThat())?

**Comment by wKich** [comment]
@ostapwd added a fallback `expectInteractor`. According cypress code they are trying to restrict users from overwriting built-in commands. So I think we need to do a breaking change and rename `cy.expect` to something else. In this PR I just added the fallback.

@cowboyd fyi

**Pull request merged**